File: SPCode/UI/Form/DockableForm.vb
Namespace UI
    Public Class DockableForm
        Inherits WeifenLuo.WinFormsUI.DockContent

        Protected Overrides Sub OnLoad(ByVal e As System.EventArgs)

            Me.TabText = Me.Text
            Try
                If Not DesignMode Then
                    If Me.MdiParent IsNot Nothing Then
                        Me.Show(CType(Me.MdiParent, DockingMDI).DockPanel, WeifenLuo.WinFormsUI.DockState.Document)
                    End If
                End If
            Catch ex As Exception
            End Try

            MyBase.OnLoad(e)
        End Sub

        Private Sub mnuClose_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles mnuClose.Click
            Me.Close()
        End Sub
    End Class
End Namespace

File: SPCode/CS/UI/Form/DockableForm.cs
using System.Text.RegularExpressions;
using System.Diagnostics;
using System;
using System.Drawing.Drawing2D;
using System.Windows.Forms;
using System.Collections;
using System.Drawing;
using Microsoft.VisualBasic;
using System.Data.SqlClient;
using System.Data;
using System.Collections.Generic;
using WeifenLuo.WinFormsUI;
using Microsoft.Win32;
using WeifenLuo;

namespace SoftLogik.Win
{
	namespace UI
	{
		public partial class DockableForm : WeifenLuo.WinFormsUI.DockContent
		{
			public DockableForm()
			{
				InitializeComponent();
			}
			
			protected override void OnLoad(System.EventArgs e)
			{
				
				this.TabText = this.Text;
				try
				{
					if (! DesignMode)
					{
						if (this.MdiParent != null)
						{
							this.Show(((DockingMDI) this.MdiParent).DockPanel, WeifenLuo.WinFormsUI.DockState.Document);
						}
					}
				}
				catch (Exception)
				{
				}
				
				base.OnLoad(e);
			}
			
			public void mnuClose_Click(System.Object sender, System.EventArgs e)
			{
				this.Close();
			}
		}
	}
	
}
